### DISCUSSION TITLE
High-level control using Vicon issue: The crazyflie flipped immediately after takeoff

### DISCUSSION THREAD

**Discussion by GuiliangZ** [bitcraze/discussions#1601] [Q&A]
Hi,

I am trying to integrate Vicon positional info with crazy fly using the crazy-flie-lib-python library by modifying the "mocap_hl_command.py". I confirmed that the Vicon data has been sent to the crazy fly and fused with the onboard sensor. I checked the "stateEstimate.x, stateEstimate.y, stateEstimate.z" which pretty much synced with the vicon position data. However, when I send any trajectory or setpoint, the crazy fly just flipped after take off. Can some one help me with the reason why and how to debug it? 

Thank you!

**Comment by gemenerik** [comment]
Hi Guiliang. Based on your description, and assuming your Vicon system is calibrated properly, one common cause for this issue is a mismatch in coordinate systems. To resolve this, please verify that the axes of the rigid body you are tracking (the Crazyflie) in the Vicon software align with the axes defined in [the Crazyflie 2.x coordinate system](https://www.bitcraze.io/documentation/system/platform/cf2-coordinate-system/).

**Reply by GuiliangZ** [reply]
Hi Gemenerik,
Thank you for your response. I think it makes a lot of sense that it could be the coordinate mismatch issue. I am wondering what would be the best way to adjust so that the coordinate system matches? Should I flip the sign at the setpoint or at the location where the Vicon data is being sent to  EKF? 
Thank you for your help!

**Reply by gemenerik** [reply]
It’s likely easiest to make these adjustments in the Vicon software. While I’m not familiar with the specifics of defining objects to track in a Vicon system, there should be a way to view and reorient the axes of the rigid body you are tracking.

**Reply by GuiliangZ** [reply]
Hi Gemenerik,
Thanks for the suggestion. However, I have checked the coordinates, and it seems the movement of the drone matches with the positional information of the Vicon - the coordinate matches. The drone is still doing the flip after it takes off. I am wondering what could be other causes of the issue? Also, how can I set the max yaw/pitch rate in the crazyflie-python-lib? Thank you for your help!

**Reply by Bananazooka** [reply]
There are several things you can try out here:
- Make sure hardware wise everything is okay, meaning that you can control the Crazyflie manually (through cfclient or the app)
- Are you sure the full coordinate system is compatible? You'll need to look at not only position, but also orientation, so full pose. You can find the supported coordinate system of the Crazyflie [here](https://www.bitcraze.io/documentation/system/platform/cf2-coordinate-system/) (pitch is inverted)
- This script is using an external python library called [motioncapture](https://github.com/IMRCLab/libmotioncapture). Maybe you can try to debug it there as well and try out some of their basic examples.

Could you ask the max yaw/pitch rate in a different thread? This is another topic and it's best to keep topics/questions separate to keep it manageable for us
